torchdata/datapipes/iter/util/webdataset.py

```diff
@@ -0,0 +1,69 @@
+import re
+from typing import Dict, Iterator, List, Union
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+def pathsplit(p):
+    """Split a path into the basename and the extensions."""
+    if "." not in p:
+        return p, ""
+    prefix, suffix = re.search(r"^(.*?)(\.[^/]*)$", p).groups()
+    return prefix, suffix
+
+
+@functional_datapipe("webdataset")
+class WebDatasetIterDataPipe(IterDataPipe[Dict]):
+    r"""
+    Iterable DataPipe that accepts stream of (path, data) tuples (usually,
+    representing the pathnames and files of a tar archive) and aggregates
+    consecutive items with the same basename into a single dictionary, using the
+    extensions as keys (WebDataset file convention). Any text after the first
```

Comment:
```suggestion
    extensions as keys based on the WebDataset file convention (functional name: ``webdataset``). Any text after the first
```

In our [documentation](https://pytorch.org/data/beta/torchdata.datapipes.iter.html), we want to show the functional name of each DataPipe in the summary table. In order for it to render properly, we typically add `(function name: ...)` at the end of the first sentence in the docstring, hence this suggestion.

Reply:
OK, I'll have a look.

Reply:
Done.